triple-store/triple-store-impl-rdf4j/src/main/java/com/powsybl/triplestore/impl/rdf4j/TripleStoreRDF4J.java

```diff
@@ -182,6 +190,38 @@ public PropertyBags query(String query) {
         return results;
     }
 
+    @Override
+    public void copyFrom(TripleStore source) {
+        Repository sourceRepo = ((TripleStoreRDF4J) source).getRepository();
+        try (RepositoryConnection sourceConn = sourceRepo.getConnection()) {
+            try (RepositoryConnection targetConn = repo.getConnection()) {
+                copyNamespaces(sourceConn, targetConn);
+                // copy statements
+                RepositoryResult<Resource> contexts = sourceConn.getContextIDs();
+                while (contexts.hasNext()) {
+                    Resource sourceContext = contexts.next();
+                    Resource targetContext = context(targetConn, sourceContext.stringValue());
+
+                    RepositoryResult<Statement> statements;
+                    statements = sourceConn.getStatements(null, null, null, sourceContext);
+                    // add statements to the new repository
+                    while (statements.hasNext()) {
+                        Statement statement = statements.next();
+                        targetConn.add(statement, targetContext);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void copyNamespaces(RepositoryConnection sourceConn, RepositoryConnection targetConn) {
+        RepositoryResult<Namespace> ns = sourceConn.getNamespaces();
+        while (ns.hasNext()) {
+            Namespace namespace = ns.next();
+            targetConn.setNamespace(namespace.getPrefix(), namespace.getName());
+        }
+    }
```

Comment:
It seems this method is duplicated: Is it possible to share the code by putting it somewhere else?

Reply:
This is not a real duplication: the two methods have different target types.